packages/manager/apps/web/client/app/hosting/general-informations/GENERAL_INFORMATIONS.controller.js

```diff
@@ -36,13 +36,18 @@ export default class HostingGeneralInformationsCtrl {
     this.HostingRuntimes = HostingRuntimes;
     this.hostingSSLCertificate = hostingSSLCertificate;
     this.OvhApiScreenshot = OvhApiScreenshot;
+
+    this.goToDetachEmail = $scope.$parent.$ctrl.goToDetachEmail;
   }
 
   $onInit() {
     this.serviceName = this.$stateParams.productId;
     this.defaultRuntime = null;
     this.availableOffers = [];
 
+    this.isDetachEmailOptionAvailable =
+      this.$scope.$parent.$ctrl.emailOptionDetachInformation.length > 0;
```

Comment:
Maybe instead you could add it to parent's resolve and inject it through constructor

Reply:
No, this controller is inject into HTML, so it is unaware of what is injected into states routing. To enable this, the hosting module must be refactor to use module imports, and then make a routing injection to this controller.